soc/silabs/silabs_s2/soc.c

```diff
@@ -39,3 +52,53 @@ void soc_early_init_hook(void)
 	sl_hfxo_manager_init();
 #endif
 }
+
+#if defined(CONFIG_ARM_SECURE_FIRMWARE) && !defined(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
```

Comment:
This `#if` should be used only if the inner code does not compile.

(If `gcc` complains with `warning: ‘smu_fault’ defined but not used`, it means `IRQ_DIRECT_CONNECT()`is not properly defined)